### PR TITLE
Homie refactor and IP/MAC address for firmware extension

### PR DESCRIPTION
**Pull request by qwandor-google**
This is mostly a refactor of the Homie library, and moving it into a separate crate.

It also adds the IP address and MAC address as required by the firmware extension. I couldn't find a way to figure out which interface is actually being used for the MQTT connection, so it just uses the default. Hopefully this is usually fine in practice.

### CODE REVIEW

File: homie/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+name = "homie"
+version = "0.1.0"
+authors = ["Andrew Walbran <qwandor@google.com>", "David Laban <alsuren@gmail.com>"]
+edition = "2018"
+
+[dependencies]
+async-channel = "1.4.0"
+futures = "*"
+local_ipaddress = "0.1"
+log = "*"
```
Comment (alsuren) [inline homie/Cargo.toml:11]: (optional) Probably worth pinning these versions down if we ever publish on crates.io. For now we can rely on Cargo.lock to pin things for us, but this won't work when we publish, I don't think.

File: homie/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+mac_address = "1.0"
+rumqttc = { git = "https://github.com/bytebeamio/rumqtt" }
```
Comment (alsuren) [inline homie/Cargo.toml:13]: have they made a release with your changes yet?

**Review by alsuren — APPROVED** [review]

File: homie/Cargo.toml (hunk 2 above)
Comment (qwandor-google) [reply homie/Cargo.toml:13]: Looks like it. Moved to that in #19

File: homie/Cargo.toml (hunk 1 above)
Comment (qwandor-google) [reply homie/Cargo.toml:11]: Done in #19.

**Pull request merged**